modules/product/src/Entity/Product.php

```diff
@@ -337,6 +337,23 @@ public static function baseFieldDefinitions(EntityTypeInterface $entityType) {
         ),
       ));
 
+    $fields['store_id'] = BaseFieldDefinition::create('entity_reference')
+      ->setLabel(t('Store'))
+      ->setDescription(t('Store field for products'))
+      ->setCardinality(1)
```

Comment:
So in the multi store marketplace where stores can sell the products, the product is "owned" by the store that created it - but we have another reference field/entity that allows other stores to sell it (this makes sense as it would also need their price on it).